packages/dds/map/src/map.ts

```diff
@@ -293,7 +294,14 @@ export class SharedMap extends SharedObject<ISharedMapEvents> implements IShared
 	): void {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
 		if (message.type === MessageType.Operation) {
-			this.kernel.tryProcessMessage(message.contents as IMapOperation, local, localOpMetadata);
+			assert(
```

Comment:
Since these errors are intended to be user-facing, and not just for internal validation, should we not explicitly throw errors, rather than asserting?

Reply:
That was also my thought, but since internal bugs could also hit this, @Abe27342 suggested we go with asserts to be consistent with our existing code and avoid possibly attributing our own bugs to users.

There really should be a package version or format version based compatibility check much earlier which would be a UsageError, but we currently have no such checking.

Reply:
Makes sense.